src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,169 @@
+     );
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            string id = diagnostic.Id;
+            switch (id)
+            {
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Logging properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Logging")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Services properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Services")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Configuration properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Configuration")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                            diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, IdentifierNameSyntax identifierMethod)
+    {
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        //whole invocation starting from "builder"
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); 
+
+        if (diagnosticTarget is InvocationExpressionSyntax originalInvocation)
```

Comment:
```suggestion
        if (diagnosticTarget is InvocationExpressionSyntax invocation)
```

Let's just call this `invocation`. It's really confusing that we call it `originalInvocation` considering that it is modified. I think it's easy enough to follow that this is the invocation that comes from the diagnostic target.